be/src/exec/parquet_builder.h

```diff
@@ -85,26 +60,12 @@ class ParquetBuilder : public FileBuilder {
 
     Status finish() override;
 
-private:
-    Status _init(const ParquetBuilderOptions& options, const std::vector<std::string>& file_column_names);
-    void _init_properties(const ParquetBuilderOptions& options);
-    Status _init_schema(const std::vector<std::string>& file_column_names);
-    void _generate_rg_writer();
-    void _flush_row_group();
-    size_t _get_rg_written_bytes();
-    void _check_size();
+    static std::shared_ptr<::parquet::WriterProperties> get_properties(const ParquetBuilderOptions& options);
+    static std::shared_ptr<::parquet::schema::GroupNode> get_schema(const std::vector<std::string>& file_column_names,
+                                                                    const std::vector<ExprContext*>& output_expr_ctxs);
 
-    std::unique_ptr<WritableFile> _writable_file;
-    std::shared_ptr<ParquetOutputStream> _output_stream;
-    const std::vector<ExprContext*>& _output_expr_ctxs;
-    std::shared_ptr<::parquet::schema::GroupNode> _schema;
-    std::shared_ptr<::parquet::WriterProperties> _properties;
-    std::unique_ptr<::parquet::ParquetFileWriter> _file_writer;
-    ::parquet::RowGroupWriter* _rg_writer = nullptr;
-    std::vector<int64_t> _buffered_values_estimate;
-    const int64_t _row_group_max_size;
-    bool _closed = false;
-    int64_t _total_row_group_writen_bytes{0};
+private:
+    std::unique_ptr<starrocks::parquet::SyncFileWriter> _writer;
```

Comment:
why `syncFileWriter` here? Is there any parent class for `SyncFileWriter` and `AsyncFileWriter`？

Reply:
we just need syncFileWriter for this case